test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/jbpm/ProcessFailoverIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+import org.slf4j.LoggerFactory;
+
+public class ProcessFailoverIntegrationTest extends AbstractCloudIntegrationTest<WorkbenchWithKieServerScenario> {
+
+    protected KieServerMgmtControllerClient kieServerMgmtControllerClient;
+    protected KieServicesClient kieServicesClient;
+    protected ProcessServicesClient processServicesClient;
+    protected QueryServicesClient queryServicesClient;
+    private static final Logger logger = LoggerFactory.getLogger(ProcessFailoverIntegrationTest.class);
+
+    private static final String variableKey = "name";
+    private static final String variableValueOne = "ONE";
+    private static final String variableValueTwo = "TWO";
+
+    @Override
+    protected WorkbenchWithKieServerScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder().build();
+    }
+
+    @Before
+    public void setUp() {
+        WorkbenchUtils.deployProjectToWorkbench(gitProvider, deploymentScenario.getWorkbenchDeployment(), DEFINITION_PROJECT_NAME);
+
+        kieServerMgmtControllerClient = KieServerControllerClientProvider.getKieServerMgmtControllerClient(deploymentScenario.getWorkbenchDeployment());
+
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+        queryServicesClient = KieServerClientProvider.getQueryClient(deploymentScenario.getKieServerDeployment());
+    }
+
+    @Test
+    public void processFailoverTest() {
+        logger.debug("Register Kie Container to Kie Server");
+        KieServerInfo serverInfo = kieServicesClient.getServerInfo().getResult();
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, DEFINITION_PROJECT_NAME, DEFINITION_PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        logger.debug("Get Kie Server Instance");
+        Instance kieServerInstance = deploymentScenario.getKieServerDeployment().getInstances().iterator().next();
+
+        logger.debug("Start process instance");
+        Long longScriptPid = processServicesClient.startProcess(CONTAINER_ID, LONG_SCRIPT_PROCESS_ID, Collections.EMPTY_MAP);
+        assertThat(longScriptPid).isNotNull().isGreaterThan(0L);
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+        assertProcessVariable(longScriptPid, variableKey, variableValueOne);
+
+        logger.debug("Send signal to continue with process.");
+        signalStartLongScript(longScriptPid);
```

Comment:
Executing of this request in separate thread was added into signalStartLongScript method.